spring-vault-core/src/main/java/org/springframework/vault/support/VaultMetadataResponse.java

```diff
@@ -0,0 +1,184 @@
+package org.springframework.vault.support;
+
+import java.time.Instant;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * Value object to bind Vault HTTP kv read metadata API responses.
+ *
+ * @author Zakaria Amine
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class VaultMetadataResponse {
```

Comment:
As mentioned earlier, let's perform deserialization of this type ourselves. That way, we get the most control over the object and we can reuse `Versioned.Metadata` for `versions` instead of exposing a Jackson-specific type.

We should make this type also immutable because the response object does not change once constructed.